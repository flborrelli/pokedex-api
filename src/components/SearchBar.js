import React from "react";
import { Input } from 'semantic-ui-react';

function SearchBar({ handleInputChange }) {
  
  return (
    <div className='search-bar mt-5'>
      <div class="ui icon input">
      <input onChange={handleInputChange} type="text" placeholder="Search a Pokemon..."/>
      <i aria-hidden="true" class="search icon"></i>
      </div>
    </div>
  );
}

export default SearchBar;