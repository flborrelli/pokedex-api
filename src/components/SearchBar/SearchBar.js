import React from "react";

function SearchBar({ handleInputChange }) {
  
  return (
    <div className='search-bar mt-5'>
      <div className="ui icon input">
      <input onChange={handleInputChange} type="text" placeholder="Search a Pokemon..."/>
      <i aria-hidden="true" className="search icon"></i>
      </div>
    </div>
  );
}

export default SearchBar;