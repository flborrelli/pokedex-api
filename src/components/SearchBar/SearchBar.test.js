import React from 'react';
import SearchBar from './SearchBar';
import { create } from 'react-test-renderer';

  //Jest Mock function
  const handleInputChange = jest.fn();

  test('should match with searchbar snapshot', () => {
    const searchComponent = create(<SearchBar handleInputChange={handleInputChange} />);
    expect(searchComponent.toJSON()).toMatchSnapshot();
  });

  test('should receive handleInputChange props', () => {
    const searchComponent = create(<SearchBar handleInputChange={handleInputChange} />);
    expect(searchComponent.toTree().props.handleInputChange).toBeTruthy();
  });

  