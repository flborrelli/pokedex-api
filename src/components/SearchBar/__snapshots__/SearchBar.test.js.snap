// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchBar /> should match with searchbar snapshot 1`] = `
<div
  className="search-bar mt-5"
>
  <div
    className="ui icon input"
  >
    <input
      onChange={[MockFunction]}
      placeholder="Search a Pokemon..."
      type="text"
    />
    <i
      aria-hidden="true"
      className="search icon"
    />
  </div>
</div>
`;

exports[`should match with searchbar snapshot 1`] = `
<div
  className="search-bar mt-5"
>
  <div
    className="ui icon input"
  >
    <input
      onChange={[MockFunction]}
      placeholder="Search a Pokemon..."
      type="text"
    />
    <i
      aria-hidden="true"
      className="search icon"
    />
  </div>
</div>
`;
